dataframely/schema.py

```diff
@@ -893,6 +896,56 @@ def _requires_validation_for_reading_parquet(
             )
         return True
 
+    # ------------------------------------- IO --------------------------------------- #
+    @classmethod
+    def _write(cls, df: pl.DataFrame, io: StorageBackend, **kwargs: Any) -> None:
+        io.write_frame(df=df, serialized_schema=cls.serialize(), **kwargs)
+
+    @classmethod
+    def _sink(cls, lf: pl.LazyFrame, io: StorageBackend, **kwargs: Any) -> None:
+        io.sink_frame(lf=lf, serialized_schema=cls.serialize(), **kwargs)
+
+    @overload
+    @classmethod
+    def _read(
+        cls,
+        io: StorageBackend,
+        validation: Validation,
+        lazy: Literal[True],
+        **kwargs: Any,
+    ) -> LazyFrame[Self]: ...
+
+    @overload
+    @classmethod
+    def _read(
+        cls,
+        io: StorageBackend,
+        validation: Validation,
+        lazy: Literal[False],
+        **kwargs: Any,
+    ) -> DataFrame[Self]: ...
+
+    @classmethod
+    def _read(
+        cls, io: StorageBackend, validation: Validation, lazy: bool, **kwargs: Any
+    ) -> LazyFrame[Self] | DataFrame[Self]:
+        read_function = io.scan_frame if lazy else io.read_frame
```

Comment:
This feels weird, can we simply do an `if` block below to assign `df` and `serialized_schema`?